### ISSUE TITLE
Templates and indentation false positives

### ISSUE THREAD

**Issue by ntdesmond** [hudochenkov/postcss-styled-syntax#2]
I searched a syntax that would not report arrow functions in interpolation as [@stylelint/postcss-css-in-js](https://github.com/stylelint/postcss-css-in-js) does.

Tried your package and not only it didn't handle the interpolation properly, but reported even more errors as it didn't like the indentation. I guess it has something to do with `indentation` rule that I use in my config.

```typescript
interface ButtonProps {
  color?: 'negative' | 'positive';
  size?: 'small' | 'normal';
}

// Errors in the CSS below:
// * Expected indentation of 0 spaces  indentation
// * Unexpected unknown function "${"  function-no-unknown
const buttonFontSize = css<ButtonProps>`
  font-size: ${({ size }) => (size === 'small') ? '0.8em' : '1em'};
`;
```

## Technical details

package versions from `package.json`:
```json
    "postcss-styled-syntax": "^0.2.0",
    "stylelint": "^14.16.0",
    "stylelint-config-recommended": "^9.0.0",
    "stylelint-config-styled-components": "^0.1.1",
```

`.stylelintrc`:

```json
{
  "extends": [
    "stylelint-config-recommended",
    "stylelint-config-styled-components"
  ],
  "rules": {
    "indentation": 2,
    "declaration-block-semicolon-newline-after": "always",
    "declaration-colon-newline-after": "always-multi-line"
  },
  "overrides": [
    {
      "files": [
        "**/*{.tsx,.ts}"
      ],
      "customSyntax": "postcss-styled-syntax"
    }
  ]
}
```

---

P.S. I switched to [43081j/postcss-styled-components](https://github.com/43081j/postcss-styled-components) which does not have these issues.

**Comment by hudochenkov** [comment]
Thank you for your report!

> it didn't handle the interpolation properly

Could you explain what was the problem?

> I guess it has something to do with indentation rule that I use in my config.

You're right. There is an issue within `indentation` rule code. This issue won't be fixed, because `indentation` rule will be [deprecated in upcoming stylelint v15](https://github.com/stylelint/stylelint/blob/v15/docs/migration-guide/to-15.md).

> Unexpected unknown function "${"  function-no-unknown

Again, issue with stylelint rule. This is something I can look into, as it's also in my to-do :)

Funny fact, `postcss-styled-components` doesn't show an error, because it replace value of `font-size` and Stylelint see it as `font-size: POSTCSS_styled-components_0`.

> P.S. I switched to [43081j/postcss-styled-components](https://github.com/43081j/postcss-styled-components) which does not have these issues.

Whatever works for you. Be aware that it currently doesn't support a lot of things related to interpolations and throws parsing errors in many common styled-components use cases. But I know that author of that package works to improve support.

**Comment by hudochenkov** [comment]
For `function-no-unknown` for now you could use:

```js
'function-no-unknown': [
	true,
	{
		ignoreFunctions: [/^\${/, /^`/],
	},
],
```

I used this configuration with `@stylelint/postcss-css-in-js` and it works with this syntax.

**Comment by hudochenkov** [comment]
Opened a PR in stylelint https://github.com/stylelint/stylelint/pull/6565

**Comment by hudochenkov** [comment]
`function-no-unknown` will be fixed in stylelint v15, when it's released.

**Comment by meriouma** [comment]
Any chance the `indentation` bug gets fixed, to be compatible with https://github.com/elirasza/stylelint-stylistic or https://github.com/firefoxic/stylelint-codeguide ?

**Comment by hudochenkov** [comment]
@meriouma it's up to developers of `indentation` rule to adjust it to work with this syntax. I'm pretty sure the rule is written for standard CSS and it doesn't expect to have properties outside of rules as we have in CSS-in-JS.

**Comment by rishitank** [comment]
I am also facing the issue with indentation, currently it doesn't add an initial indent when having CSS in JS like with styled components and as a result it expects an indentation of 0 spaces initially, but it is able to indent nested CSS correctly. I am using https://github.com/stylelint-stylistic/stylelint-stylistic
